bundles/org.eclipse.jface.text/src/org/eclipse/jface/text/contentassist/AsyncCompletionProposalPopup.java

```diff
@@ -60,6 +67,22 @@ class AsyncCompletionProposalPopup extends CompletionProposalPopup {
 
 	private static final int MAX_WAIT_IN_MS= 50; // TODO make it a preference
 
+	private static final ExecutorService EXECUTOR= new ThreadPoolExecutor(
```

Comment:
I think you should explain a bit here why this is used. and why given values are chosen.